### ISSUE TITLE
Top menu buttons, html blocks in the side bar. 

### ISSUE THREAD

**Issue by perfectdarkmode1** [McShelby/hugo-theme-relearn#739]
Is there a way to put buttons on the top bar such as a Contact page button etc.? I have them in the "More section" of the sidebar but it would nice to display these on the top menu bar. 

Also, is there a way to insert an html block and/or image with a link into the sidebar? Use case would to have a few ebook covers on the side under all of my content that when they click it, it brings the user to a store or ebook page.

**Comment by McShelby** [comment]
For your first request: Did you see the docs for the topbar? Does this help? https://mcshelby.github.io/hugo-theme-relearn/basics/topbar/index.html

For the second request: I don't plan to support this. But in the end you could store almost any HTML in a slot (I guess). It's more a coincidence that those are currently all buttons.

**Comment by perfectdarkmode1** [comment]
For the top menu, I can't seem to get the top manu to change. I have added /layouts/partials/topbar/area/end.html with the following content:

```
{{- partial "topbar/button/edit.html" (dict
	"page" .
)}}
{{- partial "topbar/button/print.html" (dict
	"page" .
)}}
{{- partial "topbar/button/prev.html" (dict
	"page" .
)}}
{{- partial "topbar/button/next.html" (dict
	"page" .
)}}
{{- partial "topbar/button/more.html" (dict
	"page" .
)}}
```

This isn't to add a custom button yet, I just wanted to make sure I could see changes before diving deeper. Nothing changes when I add this file. My site still only shows print, next, and previous buttons on the end menu like it did before. Is there any other setting I need to change to enable this?

**Comment by McShelby** [comment]
> I can't seem to get the top manu to change. I have added /layouts/partials/topbar/area/end.html with the following content:

Well, this is a copy of the shipped file. I don't expect to change something in this case.

**Comment by perfectdarkmode1** [comment]
Shouldn't this show the edit button?

**Comment by McShelby** [comment]
From the docs: 

> [edit](https://github.com/McShelby/hugo-theme-relearn/blob/main/layouts/partials/topbar/button/edit.html): browses to the editable page if the editURL [parameter is set](https://mcshelby.github.io/hugo-theme-relearn/basics/configuration/index.html)

Did you?

**Comment by perfectdarkmode1** [comment]
I actually didn't check that. But There is still nore more menu either. I saw in the docs:

"you can redefine the predefined end area by adding the file [layouts/partials/topbar/area/end.html](https://github.com/McShelby/hugo-theme-relearn/blob/main/layouts/partials/topbar/area/end.html) in your installation (not in the theme itself) to remove all but the more button"

What is the difference between "in your installation" and "not the theme itself". I want to make sure this is in the right spot.

I removed everything but:

```
{{- partial "topbar/button/print.html" (dict
	"page" .
)}}
```

And the next and previous buttons are still there. I just feel like the changes I am making aren't effecting the site at all.

**Comment by McShelby** [comment]
> What is the difference between "in your installation" and "not the theme itself". I want to make sure this is in the right spot.

Hugo create a union filesystem during execution, merging your content together with the theme's directory structure as describe here: https://gohugo.io/getting-started/directory-structure/#theme-skeleton.

If you are editing theme files, you're making something wrong.

For further help, I need to see your code: https://discourse.gohugo.io/t/requesting-help/9132#let-us-see-your-code-8

**Comment by McShelby** [comment]
The file layout should look something like this. As you can see, I removed all but the more-button from the end area and the page reflects that fact:

![grafik](https://github.com/McShelby/hugo-theme-relearn/assets/11670870/75d5d7ee-a516-4a6e-824b-7cef2a141c63)

**Comment by perfectdarkmode1** [comment]
Here is the folder layout 
![image](https://github.com/McShelby/hugo-theme-relearn/assets/60457914/3e0b52d2-d3aa-48f2-8b87-f9a869448405)

Here is what is showing: 
![image](https://github.com/McShelby/hugo-theme-relearn/assets/60457914/b69a43fe-0670-4777-b1e7-98989df5519e)

**Comment by McShelby** [comment]
The file structure looks okay. So:

> For further help, I need to see your code: https://discourse.gohugo.io/t/requesting-help/9132#let-us-see-your-code-8

**Comment by perfectdarkmode1** [comment]
Oh sorry, Here is the Github page with the site: https://github.com/perfectdarkmode1/perfectdarkmode1.github.io

**Comment by McShelby** [comment]
Your theme version is outdated (5.17.1). The topbar modification was introduced in 5.21.0: https://mcshelby.github.io/hugo-theme-relearn/basics/migration/index.html#5210

![grafik](https://github.com/McShelby/hugo-theme-relearn/assets/11670870/98593ddf-57d2-445b-a091-85ca39eed9a8)

**Comment by McShelby** [comment]
Also in 5.20.0 the zen variants got extra dividers between menu and topbar which you might find interesting.

**Comment by perfectdarkmode1** [comment]
DO you have any example on how I would set a top bar button to say "About" then have it link to an about page? I can't get custom buttons to work :(

**Comment by McShelby** [comment]
That's currently not possible but I can add a `title` attribute with the next release.

**Comment by McShelby** [comment]
I ve pushed a change:

The topbar button received a way to add text next to the icon. For this, the original `title` option was renamed to `hint` while the new `title` option is now displayed next to the icon.

**Comment by perfectdarkmode1** [comment]
I really appreciate all of your help. I am having trouble getting the custom button to display. Under layouts/partials/button/ I have a file called contact.html with the following content:

```
{{- $onempty := cond (isset . "onempty") .onempty "show" }}
{{- $onwidths := cond (isset . "onwidths") .onwidths "show" }}
{{- $onwidthm := cond (isset . "onwidthm") .onwidthm "show" }}
{{- $onwidthl := cond (isset . "onwidthl") .onwidthl "show" }}
{{- partial "topbar/func/area-button.html" (dict
	"page" .
	"area" "end"
	"icon" "envelope"
	"title" "Contact"
	"href" "https://perfectdarkmode.com/contact/"
	"onempty" $onempty
	"onwidths" $onwidths
	"onwidthm" $onwidthm
	"onwidthl" $onwidthl
	"hint" (printf "%s" (T "Contact"))
)}}
```

And I am calling it under layouts/partials/area/end.html like this:

```
{{- partial "topbar/button/print.html" (dict
	"page" .
)}}

{{- partial "topbar/button/contact.html" (dict
	"page" .
)}}
```

I tried to copy the other templates the best I could but I'm still missing something to get the button to display. Currently, just the print button displays in the end area.

**Comment by McShelby** [comment]
Two things:

1. From the docs:

> class: Mandatory unique class name for this button. Displaying two buttons with the same value for class is undefined.

2. Don't use an area button. Those are meant to show an overlay area on click like the toc or the more button. As far as I understood, you just want to link somewhere else. So just use a normal button.

tldr; Change your `contact.html` to this:

````
{{- $onempty := cond (isset . "onempty") .onempty "show" }}
{{- $onwidths := cond (isset . "onwidths") .onwidths "show" }}
{{- $onwidthm := cond (isset . "onwidthm") .onwidthm "show" }}
{{- $onwidthl := cond (isset . "onwidthl") .onwidthl "show" }}
{{- partial "topbar/func/button.html" (dict
	"page" .
	"class" "topbar-button-contact"
	"icon" "envelope"
	"title" "Contact"
	"href" "https://perfectdarkmode.com/contact/"
	"onempty" $onempty
	"onwidths" $onwidths
	"onwidthm" $onwidthm
	"onwidthl" $onwidthl
	"hint" (printf "%s" (T "Contact"))
)}}
````

**Comment by perfectdarkmode1** [comment]
This is EPIC! Thank you soo much :) Got it working.

**Comment by perfectdarkmode1** [comment]
Is the side menu linked to the "start" area at all? My sidebar on mobile no longer works. Can see it here: https://perfectdarkmode.com

**Comment by McShelby** [comment]
You need the `sidebar` button in the `start` area. Otherwise you can not access the flyout menu on mobile
